art/attacks/evasion/imperceptible_asr/imperceptible_asr_pytorch.py

```diff
@@ -259,6 +261,16 @@ class only supports targeted attack.
             # First reset delta
             self.global_optimal_delta.data = torch.zeros(self.batch_size, self.global_max_length).type(torch.float32)
 
+            # Next, reset non-SGD optimizers
+            if self.optimizer_1st_stage_arg is not None:
+                self.optimizer_1st_stage = self.optimizer_1st_stage_arg(
+                    params=[self.global_optimal_delta], lr=self.learning_rate_1st_stage
+                )
+            if self.optimizer_2nd_stage_arg is not None:
+                self.optimizer_2nd_stage = self.optimizer_2nd_stage_arg(
+                    params=[self.global_optimal_delta], lr=self.learning_rate_2nd_stage
+                )
```

Comment:
```suggestion
            # Next, reset non-SGD optimizers
            if self._optimizer_1st_stage_arg is not None:
                self.optimizer_1st_stage = self._optimizer_1st_stage_arg(
                    params=[self.global_optimal_delta], lr=self.learning_rate_1st_stage
                )
            if self._optimizer_2nd_stage_arg is not None:
                self.optimizer_2nd_stage = self._optimizer_2nd_stage_arg(
                    params=[self.global_optimal_delta], lr=self.learning_rate_2nd_stage
                )
```

Reply:
Great catch Beat! This would have created some troubles otherwise ;)